lib/util/historical-buffer-manager.ts

```diff
@@ -0,0 +1,95 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import assert = require('assert');
+import interfaces = require('../interface');
+
+// Internal class that holds the buffer array, a counter for overflow items and maxBufferLength
+class Buffer<T> {
+    data: T[];
+    overflow: number;
+    maxBufferLength: number;
+
+    get getObject(): interfaces.IBufferedData<T> {
+        return { buffer: this.data.slice(),
```

Comment:
I'll update the style guide on a section regarding object literals on this consensus.